### ISSUE TITLE
Plans for v3.5.9 release

### ISSUE THREAD

**Issue by serathius** [etcd-io/etcd#15871]
### What would you like to be added?

Issues/PRs planned for the release:
* [x] https://github.com/etcd-io/etcd/pull/15821 
  * [x] https://github.com/etcd-io/etcd/pull/15822
* [x] https://github.com/etcd-io/etcd/pull/15821
  * [x] https://github.com/etcd-io/etcd/pull/15815

Issues/PRs that can be postponed to next release:
* https://github.com/etcd-io/etcd/issues/15402 - blocked on https://github.com/golang/go/issues/58804
* https://github.com/etcd-io/etcd/issues/14653
* https://github.com/etcd-io/etcd/issues/14631 
* https://github.com/etcd-io/etcd/issues/14362
* https://github.com/etcd-io/etcd/issues/14252
* https://github.com/etcd-io/etcd/issues/14249
* https://github.com/etcd-io/etcd/issues/14143
* https://github.com/etcd-io/etcd/issues/14140
* https://github.com/etcd-io/etcd/issues/13990

### Why is this needed?

go1.19.9 (released 2023-05-02) includes three security fixes to the html/template package, as well as bug fixes to the compiler, the runtime, and the crypto/tls and syscall packages. See the [Go 1.19.9 milestone](https://github.com/golang/go/issues?q=milestone%3AGo1.19.9+label%3ACherryPickApproved) on our issue tracker for details.

**Comment by serathius** [comment]
cc @ahrtr @ptabor @spzala @mitake

**Comment by ahrtr** [comment]
My understanding is https://github.com/etcd-io/etcd/issues/15402 is already resolved in 3.5.8 by splitting the http and gRPC servers. Eventually in 3.7 or 4.0, etcd will split them by default. 

Do you mean that etcd should follow up the http server issue, which is supposed to be resolved by Go team, so that the issue can be resolved even users of etcd do not split http and gRPC servers in etcd 3.4 and 3.5?

**Comment by serathius** [comment]
I mean https://github.com/golang/go/issues/58804 is fixed.

**Comment by serathius** [comment]
> Do you mean that etcd should follow up the http server issue, which is supposed to be resolved by Go team, so that the issue can be resolved even users of etcd do not split http and gRPC servers in etcd 3.4 and 3.5?

Yes, we should not require that all users follow etcd changelog and know all the flags. We should provide a proper solution available by default.

**Comment by serathius** [comment]
Published https://github.com/etcd-io/etcd/releases/tag/v3.5.9
